### DISCUSSION TITLE
有没有好的Java搜索思路

### DISCUSSION THREAD

**Discussion by SALTWOOD** [Meloong-Git/PCL#2398] [讨论]
![1690861197104](https://github.com/Hex-Dragon/PCL2/assets/105980161/bcd7d30e-2e65-49de-a96f-3b08b5bf7653)

最近在写一些废柴，涉及到Java的搜索
就想到来扒一扒PCL2的源码
感觉Java搜索很乱索性自己写
有哪位大神有好的思路吗

**Comment by hejiehao** [comment]
扫环境变量，扫盘？

**Reply by SALTWOOD** [reply]
我看PCL2是扫环境变量、硬盘（关键字匹配，比如带有“Java”字样的文件夹）、AppData
也许可以学一下Everything...?
话说Everything开源吗*

**Reply by Super-XiaoNian** [reply]
Everything好像是第一次启动就扫描硬盘，然后创建个索引数据包对应文件，这样每次搜索都去索引数据库搜索

**Reply by SALTWOOD** [reply]
在GitHub.com搜索到一个仓库做了快速扫盘的原理分析，利用的是NTFS的USN特性，想着现在用NTFS的人也比较多，那就试试吧

谢谢大佬
